components/eam/src/physics/cam/hetfrz_classnuc_cam.F90

```diff
@@ -876,368 +890,383 @@ end subroutine hetfrz_classnuc_cam_save_cbaero
+   !   
+   ! The 6.0/(dgncur_a*fac_volsfc_dust) = (mode-surface-area/mode-volume)
+   ! Note that vol_shell includes both so4, pom, AND soa as "equivalent so4",
+   ! The soa_equivso4_factor accounts for the lower hygroscopicity of soa.
+   !
+   ! Define xferfrac_pcage = min(1.0, ratio1/ratio2)
+   !   But ratio1/ratio2 == tmp1/tmp2, and coding below avoids possible overflow 
+
+   ! bc
+   fac_volsfc_bc      = exp(2.5_r8*alnsg_mode_pcarbon**2)
+
+   vol_shell(1) = ( aer(ii,kk,pom_pcarbon)*pom_equivso4_factor/specdens_pom + &
+                    aer(ii,kk,mom_pcarbon)*mom_equivso4_factor/specdens_mom )/rhoair
+
+   vol_core(1)  = aer(ii,kk,bc_pcarbon)/(specdens_bc*rhoair)
+   coat_ratio1 = vol_shell(1)*(r_bc*2._r8)*fac_volsfc_bc
+   coat_ratio2 = max(6.0_r8*dr_so4_monolayers_dust*vol_core(1), 0.0_r8)
+   dstcoat(1) = coat_ratio1/coat_ratio2
+
+   ! dust_a1
+   coat_ratio1 = vol_shell(2)*(r_dust_a1*2._r8)*fac_volsfc_dust_a1
+   coat_ratio2 = max(6.0_r8*dr_so4_monolayers_dust*vol_core(2), 0.0_r8)
+   dstcoat(2) = coat_ratio1/coat_ratio2
+
+   ! dust_a3
+   vol_shell(3) = aer(ii,kk,so4_coarse)/(specdens_so4*rhoair) + & 
+                  aer(ii,kk,pom_coarse)/(specdens_pom*rhoair) + & 
+                  aer(ii,kk,soa_coarse)/(specdens_soa*rhoair) + & 
+                  aer(ii,kk,mom_coarse)/(specdens_mom*rhoair)
+
+   vol_core(3)  = aer(ii,kk,dst_coarse)/(specdens_dust*rhoair)
+   coat_ratio1 = vol_shell(3)*(r_dust_a3*2._r8)*fac_volsfc_dust_a3
+   coat_ratio2 = max(6.0_r8*dr_so4_monolayers_dust*vol_core(3), 0.0_r8)
+   dstcoat(3) = coat_ratio1/coat_ratio2
+      
+   dstcoat(1) = min_max_bound(0.001_r8, 1.0_r8, dstcoat(1))
+   dstcoat(2) = min_max_bound(0.001_r8, 1.0_r8, dstcoat(2))
+   dstcoat(3) = min_max_bound(0.001_r8, 1.0_r8, dstcoat(3))
+
+
+   do ispec = 1, 3
+      total_aer_num(ispec)    = total_interstitial_aer_num(ispec) + total_cloudborne_aer_num(ispec)
+      coated_aer_num(ispec)   = total_interstitial_aer_num(ispec)*dstcoat(ispec)
+      uncoated_aer_num(ispec) = total_interstitial_aer_num(ispec)*(1._r8-dstcoat(ispec))
+   enddo
+
+   coated_aer_num(1)   = aer(ii,kk,bc_pcarbon)*bc_kg_to_num*num_m3_to_cm3*dstcoat(1)+ &
+                         aer(ii,kk,bc_accum)*bc_kg_to_num*num_m3_to_cm3
+
+   uncoated_aer_num(1) = aer(ii,kk,bc_pcarbon)*bc_kg_to_num*num_m3_to_cm3*(1._r8-dstcoat(1))
+   
 
-      if (awcam(3) > 0._r8) then
-         awfacm(3) = ( aer(ii,kk,bc_coarse) + aer(ii,kk,soa_coarse) + &
-                       aer(ii,kk,pom_coarse) + aer(ii,kk,mom_coarse) )/ &
-                     ( aer(ii,kk,soa_coarse) + aer(ii,kk,pom_coarse) + &
-                       aer(ii,kk,so4_coarse) + aer(ii,kk,bc_coarse) + aer(ii,kk,mom_coarse) )
-      else
-         awfacm(3) = 0._r8
-      end if 
+   tot_na500 = total_aer_num(1)*0.0256_r8 + &   ! scaled for D>0.5 um using Clarke et al., 1997; 2004; 2007: rg=0.1um, sig=1.6
+               total_aer_num(2)*dst1_scale + &
+               total_aer_num(3)
```

Comment:
It is up to you but this calculation can also be a function as it is the same as the `na500` calculation below.

Reply:
I decided to leave them for now. The calculation is quite simple here.

Reply:
Would you please make `0.0256_r8` a parameter variable as it is used at multiple places?